src/arrays.js

```diff
@@ -8,34 +8,62 @@ const each = (elements, cb) => {
   // Iterates over a list of elements, yielding each in turn to the `cb` function.
   // This only needs to work with arrays.
   // based off http://underscorejs.org/#each
+  for (let i = 0; i < elements.length; i++) { cb(elements[i], i); }
 };
 
 const map = (elements, cb) => {
   // Produces a new array of values by mapping each value in list through a transformation function (iteratee).
   // Return the new array.
+  const newArr = [];
+  for (let i = 0; i < elements.length; i++) { newArr.push(cb(elements[i])); }
+  return newArr;
 };
 
 const reduce = (elements, cb, memo = elements.shift()) => {
   // Combine all elements into a single value going from left to right.
   // Elements will be passed one by one into `cb`.
-  // `memo` is the starting value.  If `memo` is undefined then make `elements[0]` the initial value.
+  // `memo` is the starting value.  If `memo` is undefined then make `elements[0]` the initial value
+  if (memo === undefined) memo = elements[0];
+  for (let i = 0; i < elements.length; ++i) {
+    memo = cb(memo, elements[i]);
+  }
+  return memo;
 };
 
+
 const find = (elements, cb) => {
   // Look through each value in `elements` and pass each element to `cb`.
   // If `cb` returns `true` then return that element.
   // Return `undefined` if no elements pass the truth test.
+  for (let i = 0; i < elements.length; i++) {
+    if (cb(elements[i])) return elements[i];
+  }
+  return undefined;
```

Comment:
Just to save yourselves a few keystrokes, you can just `return;` here, which is the same as doing `return undefined;`.